### ISSUE TITLE
Carbon PHP 8+ deprecation errors/notices

### ISSUE THREAD

**Issue by davidsherlock** [awesomemotive/easy-digital-downloads#9625]
### Expected behavior
I expect to be able to use PHP 8+ and upgrade without issues.

### Actual behavior

I get the following errors/notices:

```
Deprecated | Return type of Carbon\Carbon::createFromFormat($format, $time, $tz = null) should either be compatible with DateTime::createFromFormat(string $format, string $datetime, ?DateTimeZone $timezone = null): DateTime\|false, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice | 1 | +wp-content/plugins/easy-digital-downloads/includes/libraries/Carbon.php:568 | Plugin: easy-digital-downloads
-- | -- | -- | -- | --
Deprecated | Return type of Carbon\Carbon::getLastErrors() should either be compatible with DateTime::getLastErrors(): array\|false, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice | 1 | +wp-content/plugins/easy-digital-downloads/includes/libraries/Carbon.php:600 | Plugin: easy-digital-downloads
Deprecated | Return type of Carbon\Carbon::setTimezone($value) should either be compatible with DateTime::setTimezone(DateTimeZone $timezone): DateTime, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice | 1 | +wp-content/plugins/easy-digital-downloads/includes/libraries/Carbon.php:944 | Plugin: easy-digital-downloads
Deprecated | Return type of Carbon\Carbon::setDate($year, $month, $day) should either be compatible with DateTime::setDate(int $year, int $month, int $day): DateTime, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice | 1 | +wp-content/plugins/easy-digital-downloads/includes/libraries/Carbon.php:859 | Plugin: easy-digital-downloads
Deprecated | Return type of EDD\Utils\Date::format($format = 'datetime') should either be compatible with DateTime::format(string $format): string, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice | 1 | +wp-content/plugins/easy-digital-downloads/includes/utils/class-date.php:58 | Plugin: easy-digital-downloads
Deprecated | DateTime::__construct(): Passing null to parameter #1 ($datetime) of type string is deprecated | 1 | +wp-content/plugins/easy-digital-downloads/includes/libraries/Carbon.php:291 | Plugin: easy-digital-downloads
```

### Steps to reproduce the behavior
1) Install PHP 8.1.9

### Information (if a specific version is affected):
PHP Version: 8.1.9

EDD Version (or branch): latest

WordPress Version: 6.2

**Comment by robincornett** [comment]
Thanks @davidsherlock this is on our roadmap, although we are keeping an eye on WordPress Core support for PHP8+; currently Core only fully supports up to PHP 7.4.

**Comment by davidsherlock** [comment]
Thanks @robincornett, makes sense. As always WordPress is miles behind tech stack wise :)

It's worth noting, I think this is pretty much the only area where I see an issue. Everything else appears to work fine EDD wise.

**Comment by cklosowski** [comment]
This has been fixed in EDD 3.2.0, we've updated Carbon in this release for PHP 8+ compatibility.
